v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/transforms/SourceWriterFn.java

```diff
@@ -208,6 +203,12 @@ public void processElement(ProcessContext c) {
       // no shard found, move to permanent error
       outputWithTag(
           c, Constants.PERMANENT_ERROR_TAG, Constants.SHARD_NOT_PRESENT_ERROR_MESSAGE, spannerRec);
+    } else if (shardId.equals(Constants.RETRYABLE_SENTINEL_SHARD_ID)) {
+      outputWithTag(
+          c, Constants.RETRYABLE_ERROR_TAG, Constants.SHARD_NOT_PRESENT_ERROR_MESSAGE, spannerRec);
+    } else if (shardId.equals(Constants.SEVERE_SENTINEL_SHARD_ID)) {
```

Comment:
code structuring nit: keep the severe errors in one block and retryable errors in one block to make it clearer when the transition is being made in the code. In this context, set the severe error tag before the retryable error tag.